src/common/io.ts

```diff
@@ -154,38 +154,48 @@ export class ARIOReadable implements AoARIORead {
     });
   }
 
+  private async computeEpochIndexForTimestamp(
+    timestamp: number,
+  ): Promise<number> {
+    const epochSettings = await this.getEpochSettings();
```

Comment:
you can memoize this by adding `private epochSettings: AoEpochSettings | undefined`. set it if it has not been set, return it if it already has.

Reply:
correct, in that case it would be a forced refresh

Reply:
memoized above getEpochSettings